cmd/agentless-scanner/main.go

```diff
@@ -6,5 +6,305 @@
 // Package main implements the agentless-scanner command.
 package main
 
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"syscall"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/agentless/runner"
+	"github.com/DataDog/datadog-agent/pkg/agentless/types"
+
+	// DataDog agent: config stuffs
+	commonpath "github.com/DataDog/datadog-agent/cmd/agent/common/path"
+	"github.com/DataDog/datadog-agent/comp/core"
+	compconfig "github.com/DataDog/datadog-agent/comp/core/config"
+	complog "github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+	"github.com/DataDog/datadog-agent/comp/core/secrets"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform/eventplatformimpl"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatformreceiver/eventplatformreceiverimpl"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	"github.com/DataDog/datadog-agent/pkg/pidfile"
+	"github.com/DataDog/datadog-agent/pkg/security/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	ddogstatsd "github.com/DataDog/datadog-go/v5/statsd"
+
+	_ "modernc.org/sqlite" // sqlite driver, used by github.com/knqyf263/go-rpmdb
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultWorkersCount = 15
+	defaultScannersMax  = 8 // max number of child-process scanners spawned by a worker in parallel
+)
+
+var statsd *ddogstatsd.Client
+var eventForwarder eventplatform.Component
+
+var globalFlags struct {
+	configFilePath string
+	diskMode       types.DiskMode
+	defaultActions []types.ScanAction
+	noForkScanners bool
+}
+
+func runWithModules(run func(cmd *cobra.Command, args []string) error) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		return fxutil.OneShot(
```

Comment:
It is recommended to have one `fxutil.OneShot` per command to allow having the components only needed by the command. The current implementation forces to have the same component instantiated for each command.

Note: lines https://github.com/DataDog/datadog-agent/pull/23484/files#diff-b8a0f435748d226bbe4ceef46dc31888a5553823587d0ce1190d51f246ae65e8R75-R83 could be shared across command as they are called only when needed.

Reply:
Currently, all the commands use the `log`, `config` and `eventplatform` components.